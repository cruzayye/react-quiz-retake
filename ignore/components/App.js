import React from 'react';
import PropTypes from 'prop-types';
import ColorSelector from './ColorSelector';
import TextSelector from './TextSelector';
import Shape from './Shape';

export default class App extends React.PureComponent {
  state = {
    color: '#000000',
    backgroundColor: '#FFFFFF',
    text: ''
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  click() {
    console.log('click');
  }

  render() {
    const { color, backgroundColor, text } = this.state;
    return (
      <>
        <button onClick={this.click.bind(this)}>Click</button>
        <TextSelector text={text} onChange={this.handleChange} />
        <ColorSelector
          color={color}
          backgroundColor={backgroundColor}
          onChange={this.handleChange}
        />
        <Shape
          color={color}
          backgroundColor={backgroundColor}
          text={text}
        />
      </>
    );
  }
}
