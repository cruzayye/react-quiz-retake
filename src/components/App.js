import React, { PureComponent } from 'react';
import Controls from './Controls.js';
import Display from './Display';



export default class App extends PureComponent {
  state = {
    title: '',
    subtitle: ''

  };

  textChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  // click() {
  //   console.log('click');
  // }

  render() {
    const { title, subtitle } = this.state;
    return (
      <>
      <Controls type='text' name='title' value={title} onChange={this.textChange}
      />
      <Controls type='text'  name='subtitle' value={subtitle} onChange={this.textChange}
      />
      <Display
        first={title}
        second={subtitle}
      />
      </>
    );
  }
}


