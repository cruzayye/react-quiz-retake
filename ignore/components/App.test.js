import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';
import { wrap } from 'module';

describe('App', () => {

  it('can change text', () => {
    const wrapper = mount(<App />);

    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'click');

    wrapper.find('button').simulate('click', {
      preventDefault: jest.fn()
    });

    expect(instance.click).toHaveBeenCalled();
  });
});
