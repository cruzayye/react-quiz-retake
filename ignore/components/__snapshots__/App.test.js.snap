// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<input
  name="text"
  onChange={[MockFunction]}
  value="ryan"
/>
`;

exports[`TextSelctor matches snapshot 1`] = `
<input
  name="text"
  onChange={[MockFunction]}
  value="ryan"
/>
`;
